### PR TITLE
[Onboarding] Adds ConnectedAskName Components to Onboarding

### PR DESCRIPTION
**Pull request by cdanwards**
## DESCRIPTION

When proceeding through the onboarding process, a user will be able to enter their Name and it will be added to their profile.

### What does this PR do, or why is it needed?

This adds the `Query` and `Mutation` components to the already created `AskName` component and allows the information entered to be updated for the profile. This also adds formik and yup for form validation.

### What design trade-offs/decisions were made?

N.A.

### How do I test this PR?

You can launch the Onboarding Slides and after entering your name, you hit enter on the `lastName` and the query for updating the name will fire. You can then navigate to the connect page to see if your changes took effect.

### What automated tests did you write?

Updated tests for the `AskName` component and wrote tests for `AskNameConnected`

## TODO:

- [ ] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [ ] Closes [tag-issues-here]
- [ ] No new warnings in tests, in storybook, and in-app
- [ ] Upload GIF(s) of iOS and Android
- [ ] Set a relevant reviewer

## REVIEW:

**Manual QA**

- [x] Manual QA on iOS and ensure it looks/behaves as expected
- [x] Manual QA on Android and ensure it looks/behaves as expected

**Code Review: Questions to consider**

- [x] Read through the "Files changed" tab _very carefully_
- [x] Edge cases: what assumptions are made about input?
- [x] What kind of tests could be written?
- [ ] How might we make this easier for someone else to understand?
- [ ] Could the code be simpler?
- [ ] Will the code be easy to modify in the future?
- [ ] What's one part of these changes that makes you excited to merge it?

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

File: packages/apolloschurchapp/src/ui/Onboarding/slides/AskName/AskName.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+const StyledH5 = styled(({ theme }) => ({
+  color: theme.colors.text.secondary,
+}))(H5);
+
+// memo = sfc PureComponent 💥
+// eslint-disable-next-line react/display-name
+const AskName = memo(
+  ({
+    onPrimaryPress,
```
Comment (cdanwards) [inline packages/apolloschurchapp/src/ui/Onboarding/slides/AskName/AskName.js]: This prop can be removed
